src/Notifications/NotificationCenterController.cs

```diff
@@ -46,9 +56,45 @@ internal NotificationCenterController(DynamoView view, DynamoLogger dynLogger)
                 HorizontalOffset = notificationPopupHorizontalOffset,
                 VerticalOffset = notificationPopupVerticalOffset
             };
+
             notificationUIPopup.webView.EnsureCoreWebView2Async();
             notificationUIPopup.webView.CoreWebView2InitializationCompleted += WebView_CoreWebView2InitializationCompleted;
             logger = dynLogger;
+
+            RequestNotifications();
+        }
+
+        private void WebView_NavigationCompleted(object sender, Microsoft.Web.WebView2.Core.CoreWebView2NavigationCompletedEventArgs e)
+        {
+            AddNotifications(notificationsModel.Notifications);
+        }
+
+        private void AddNotifications(List<NotificationItemModel> notifications)
+        {
+            var notificationsList = JsonConvert.SerializeObject(notifications);
+            InvokeJS($"window.setNotifications({notificationsList});");
+        }
+
+        private void RequestNotifications()
+        {
+            var uri = DynamoUtilities.PathHelper.getServiceBackendAddress(this, "notificationAddress");
+            HttpWebRequest request = (HttpWebRequest)WebRequest.Create(uri);
+            request.AutomaticDecompression = DecompressionMethods.GZip | DecompressionMethods.Deflate;
+
+            using (HttpWebResponse response = (HttpWebResponse)request.GetResponse())
+            using (Stream stream = response.GetResponseStream())
+            using (StreamReader reader = new StreamReader(stream))
+            {
+                jsonStringFile = reader.ReadToEnd();
+                notificationsModel = JsonConvert.DeserializeObject<NotificationsModel>(jsonStringFile);
+
+                var notificationsNumber = notificationsModel.Notifications.Count();
```

Comment:
This is counting all the notifications, but we may add a comparison between the read notification and the new ones.